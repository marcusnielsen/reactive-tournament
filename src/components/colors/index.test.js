import Colors, {defaultPrimaryMainHex} from '../colors'

test('create default state', () => {
  const colors = Colors({initialPrimaryMainHex: defaultPrimaryMainHex})

  return colors.state_.take(1).forEach((actual) => {
    expect(actual).toMatchSnapshot()
  })
})

test('change color', () => {
  const colors = Colors({initialPrimaryMainHex: defaultPrimaryMainHex})

  const promise = colors.state_.skip(1).take(1).forEach((actual) => {
    expect(actual).toMatchSnapshot()
  })

  colors.actions.triggers.setPrimaryMainHex('#882D60')

  return promise
})
