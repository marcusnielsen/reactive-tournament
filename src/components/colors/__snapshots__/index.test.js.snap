exports[`test change color 1`] = `
Object {
  "complementaryMain": "hsl(146.39999999999998, 50.3%, 35.5%)",
  "complementarySecondary": "hsl(206.39999999999998, 50.3%, 35.5%)",
  "primaryMain": "hsl(326.4, 50.3%, 35.5%)",
  "primarySecondary": "hsl(26.399999999999977, 50.3%, 35.5%)",
}
`;

exports[`test create default state 1`] = `
Object {
  "complementaryMain": "hsl(235.79999999999995, 49.8%, 44.5%)",
  "complementarySecondary": "hsl(295.79999999999995, 49.8%, 44.5%)",
  "primaryMain": "hsl(55.80000000000001, 49.8%, 44.5%)",
  "primarySecondary": "hsl(115.80000000000001, 49.8%, 44.5%)",
}
`;
